enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -9785,15 +9872,24 @@ class AdjointGenerator
           IRBuilder<> Builder2(&call);
           getForwardBuilder(Builder2);
           auto origfree = orig->getArgOperand(0);
+          auto newfree = gutils->getNewFromOriginal(orig->getArgOperand(0));
           auto tofree = gutils->invertPointerM(origfree, Builder2);
-          if (tofree != origfree) {
-            SmallVector<Value *, 2> args = {tofree};
-            CallInst *CI = Builder2.CreateCall(orig->getFunctionType(),
-                                               orig->getCalledFunction(), args);
-            CI->setAttributes(orig->getAttributes());
-          }
+
+          Function *free = getOrInsertCheckedFree(
+              *orig->getModule(), orig->getFunctionType(),
```

Comment:
Can you instead pass a width argument to the insert checked free, and then pass all of the unwrapped as arguments?

That way we can check if a vector width shadow is the same as another vector width shadow?